File: app/src/main/java/com/worldpeak/chnsmilead/oa/viewmodel/OaMsgViewModel.java
package com.worldpeak.chnsmilead.oa.viewmodel;

import android.text.TextUtils;

import androidx.lifecycle.MutableLiveData;

import com.worldpeak.chnsmilead.base.BaseViewModel;
import com.worldpeak.chnsmilead.net.BaseResponse;
import com.worldpeak.chnsmilead.net.CommonRequest;
import com.worldpeak.chnsmilead.net.RetrofitListener;
import com.worldpeak.chnsmilead.oa.model.ContactOrgItem;

import java.util.List;

public class OaMsgViewModel extends BaseViewModel {

    private String TAG = "OaMsgViewModel==";
    public MutableLiveData<List<ContactOrgItem>> contactList = new MutableLiveData<>();

    public void getContactList() {
        CommonRequest.getInstance().getContactList(new RetrofitListener<BaseResponse>() {
            @Override
            public void onSuccess(BaseResponse data) {
                if (data != null && data.isSuccessful()) {
                    List<ContactOrgItem> list = data.getResultList(ContactOrgItem.class);
                    contactList.postValue(list);
                } else {
                    if (!TextUtils.isEmpty(data.getMsg())) {
                        showToast(data.getMsg());
                    }
                }
            }

            @Override
            public void onError(BaseResponse data, String description) {
                if (!TextUtils.isEmpty(description)) {
                    showToast(description);
                }
            }
        });
    }

    /**
     * 撤销
     *
     * @param formId
     */
    public void rebackNoticeDelivery(String formId) {
        CommonRequest.getInstance().rebackNoticeDelivery(formId, new RetrofitListener<BaseResponse>() {
            @Override
            public void onSuccess(BaseResponse data) {
                if (!TextUtils.isEmpty(data.getMsg())) {
                    showToast(data.getMsg());
                }
            }

            @Override
            public void onError(BaseResponse data, String description) {
                if (!TextUtils.isEmpty(description)) {
                    showToast(description);
                }
            }
        });
    }

    public void deleteNoticeDelivery(List<String> ids) {
        CommonRequest.getInstance().deleteNoticeDelivery(ids, new RetrofitListener<BaseResponse>() {
            @Override
            public void onSuccess(BaseResponse data) {
                if (!TextUtils.isEmpty(data.getMsg())) {
                    showToast(data.getMsg());
                }
            }

            @Override
            public void onError(BaseResponse data, String description) {
                if (!TextUtils.isEmpty(description)) {
                    showToast(description);
                }
            }
        });
    }

    public void remindNotConfirm(String formId) {
        if (TextUtils.isEmpty(formId)) return;
        CommonRequest.getInstance().remindNoticeNotConfirm(formId, new RetrofitListener<BaseResponse>() {
            @Override
            public void onSuccess(BaseResponse data) {
                if (data != null && data.isSuccessful()) {
                    showToast("提醒成功");
                } else {
                    if (!TextUtils.isEmpty(data.getMsg())) {
                        showToast(data.getMsg());
                    }
                }
            }

            @Override
            public void onError(BaseResponse data, String description) {
                if (!TextUtils.isEmpty(description)) {
                    showToast(description);
                }
            }
        });
    }
}

File: app/src/main/java/com/worldpeak/chnsmilead/oa/viewmodel/OaMsgViewModel.kt
package com.worldpeak.chnsmilead.oa.viewmodel

import android.text.TextUtils
import android.util.Log
import androidx.lifecycle.MutableLiveData
import com.google.gson.Gson
import com.tencent.imsdk.v2.V2TIMConversationResult
import com.tencent.imsdk.v2.V2TIMManager
import com.tencent.imsdk.v2.V2TIMValueCallback
import com.worldpeak.chnsmilead.base.BaseViewModel
import com.worldpeak.chnsmilead.net.BaseResponse
import com.worldpeak.chnsmilead.net.CommonRequest
import com.worldpeak.chnsmilead.net.RetrofitListener
import com.worldpeak.chnsmilead.oa.model.ContactOrgItem


class OaMsgViewModel : BaseViewModel() {
    var contactList = MutableLiveData<List<ContactOrgItem>>()
    private fun getContactList() {
        CommonRequest.getInstance().getContactList(object : RetrofitListener<BaseResponse<*>> {
            override fun onSuccess(data: BaseResponse<*>?) {
                if (data?.isSuccessful!!) {
                    val list = data.getResultList(
                        ContactOrgItem::class.java
                    )
                    list.map { Log.e("tag", "item = ${Gson().toJson(it)}") }
                    contactList.postValue(list)
                } else {
                    showToast(data.msg ?: "请求失败")
                }
            }

            override fun onError(data: BaseResponse<*>, description: String) {
                if (!TextUtils.isEmpty(description)) {
                    showToast(description)
                }
            }
        })
    }

    /**
     * 撤销
     *
     * @param formId
     */
    fun rebackNoticeDelivery(formId: String?) {
        CommonRequest.getInstance()
            .rebackNoticeDelivery(formId, object : RetrofitListener<BaseResponse<*>> {
                override fun onSuccess(data: BaseResponse<*>?) {
                    if (!TextUtils.isEmpty(data?.msg)) {
                        showToast(data?.msg!!)
                    }
                }

                override fun onError(data: BaseResponse<*>, description: String) {
                    if (!TextUtils.isEmpty(description)) {
                        showToast(description)
                    }
                }
            })
    }

    fun deleteNoticeDelivery(ids: List<String?>?) {
        CommonRequest.getInstance()
            .deleteNoticeDelivery(ids, object : RetrofitListener<BaseResponse<*>> {
                override fun onSuccess(data: BaseResponse<*>?) {
                    if (!TextUtils.isEmpty(data?.msg)) {
                        showToast(data?.msg!!)
                    }
                }

                override fun onError(data: BaseResponse<*>, description: String) {
                    if (!TextUtils.isEmpty(description)) {
                        showToast(description)
                    }
                }
            })
    }

    fun remindNotConfirm(formId: String?) {
        if (TextUtils.isEmpty(formId)) return
        CommonRequest.getInstance()
            .remindNoticeNotConfirm(formId, object : RetrofitListener<BaseResponse<*>> {
                override fun onSuccess(data: BaseResponse<*>?) {
                    if (data != null && data.isSuccessful) {
                        showToast("提醒成功")
                    } else {
                        if (!TextUtils.isEmpty(data?.msg)) {
                            showToast(data?.msg!!)
                        }
                    }
                }

                override fun onError(data: BaseResponse<*>, description: String) {
                    if (!TextUtils.isEmpty(description)) {
                        showToast(description)
                    }
                }
            })
    }
}